### ISSUE TITLE
State: Add computed Jetpack properties to getSite() selector

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#9122]
When a site is fetched using SitesList, this is currently decorated with a number of computed properties in the [Jetpack class](https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/jetpack.js)

So in older components we might attempt to access a computed property like `hasMinimumJetpackVersion` which isn't available on a raw site object:
```javascript
const site = sites.getSelectedSite();
if ( site.jetpack && site.hasMinimumJetpackVersion ) {
// do stuff
}
```

If we try to update the usage directly:
```javascript
const site = getSite( state, siteId );
// site.hasMinimumJetpackVersion will be undefined
```

Let's update the getSite selector to provide these computed properties:
https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/selectors.js#L67
```javascript
export const getSite = createSelector(
	( state, siteId ) => {
		const site = getRawSite( state, siteId );

		if ( ! site ) {
			return null;
		}

		return {
			...site,
			...getComputedAttributes( site ), //most likely we'll update this piece
			hasConflict: isSiteConflicting( state, siteId ),
			title: getSiteTitle( state, siteId ),
			slug: getSiteSlug( state, siteId ),
			domain: getSiteDomain( state, siteId ),
			is_previewable: isSitePreviewable( state, siteId )
		};
	},
	( state ) => state.sites.items
);
```

See also #8762 which has a number of new Jetpack selectors.

**Comment by ebinnion** [comment]
I did a survey of the selectors that we should keep for Jetpack sites. 

- `canJetpackSiteManage()`: While manage is now enabled by default as of Jetpack 4.1, sites that have Jetpack greater than 3.4 and less than 4.1 could decide to enable manage or not.
- `canJetpackSiteUpdateFiles()`: We can't update plugins/themes unless updates are allowed by a site. I don't know all of these cases off of the top of my head, but to give an example, I believe that updates fail if a site is source-controlled.
- `canJetpackSiteAutoUpdateFiles()`: Even if a site can update, it's possibly that the owner has turned off auto-updates.
- `canJetpackSiteAutoUpdateCore()`: It looks like this is the same as `canAutoupdateFiles`. Perhaps this separate method was added in anticipation that we'd need to change it for core in the future? Also, we don't currently handle core updates, so it may be nice to keep this as a separate selector for when we do implement.
- `hasJetpackSiteJetpackMenus()`: This is currently being used to determine whether we support menus on the Jetpack site, and checks that the Jetpack site has a version of at least 3.5.
- `hasJetpackSiteJetpackThemes()`: This is currently being used in one place, on the single site themes list, and does a check to make sure the Jetpack site is at least version 3.7.
- `isJetpackSiteMainNetworkSite()`: This seems to be used mainly in plugins, and could be important for determining whether to allow plugins/themes to be uploaded/activated/etc. for a given site
- `isJetpackSiteSecondaryNetworkSite()`: This is very similar to the `isJetpackSiteMainNetworkSite()` selector. This will return true if the current site is on a multisite and not the main network site. This could be useful for determining whether we should show error messaging if a site can't update/install plugins or themes.
- `verifyJetpackModulesActive()`: This selector seems to be used to determine if a module is active., which is important.
- `getJetpackSiteRemoteManagementURL()`:  This seems to be useful for building a URL to the Jetpack admin page.
- `hasJetpackSiteCustomDomain()`: I don't think we need this. Don't all Jetpack sites have custom domains?
- `getJetpackSiteFileModDisableReasons()`: This is important so we know why we can't modify files on the Jetpack sites.
- `siteHasMinimumJetpackVersion()`: This sets a boolean for determining whether Jetpack has the bare minimum version to be used with Calypso. While it looks like the minimum version is 3.3, which is several versions ago, I'm not sure we can decide to get rid of this check.

In short, I think we'll likely need to keep all of these selectors. I don't think we need to attach all of them to the site object. 

For example: For `verifyJetpackModulesActive()`, we could just wrap any component that checks whether a module is active in Redux and then pass down a boolean as a prop. As opposed to adding a callback to the site object. Perhaps, we could do something similar with `verifyJetpackModulesActive()`.

**Comment by ebinnion** [comment]
As a note, #9389 did add many of these selectors. We may need to make adjustments as we begin to use the new redux site in components. But, we may decide to not add those selectors to the site object and instead import them where needed in components.
